streams/src/test/java/org/apache/kafka/streams/kstream/internals/SessionWindowedCogroupedKStreamImplTest.java

```diff
@@ -0,0 +1,310 @@
+    public void shouldNotHaveNullMaterialized2OnAggregate() {
+        windowedCogroupedStream.aggregate(MockInitializer.STRING_INIT, Named.as("name"), sessionMerger, null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullSessionMerger3OnAggregate() {
+        windowedCogroupedStream.aggregate(MockInitializer.STRING_INIT, Named.as("name"), null, Materialized.as("test"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullNamedOnAggregate() {
+        windowedCogroupedStream.aggregate(MockInitializer.STRING_INIT, null, sessionMerger, Materialized.as("test"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullInitializer3OnAggregate() {
+        windowedCogroupedStream.aggregate(null, Named.as("name"), sessionMerger, Materialized.as("test"));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldNotHaveNullNamed2OnAggregate() {
+        windowedCogroupedStream.aggregate(MockInitializer.STRING_INIT, null, sessionMerger);
+    }
+
+
+    @Test
+    public void sessionWindowAggregateTest() {
+        final KTable<Windowed<String>, String> customers = groupedStream.cogroup(MockAggregator.TOSTRING_ADDER)
+                .windowedBy(SessionWindows.with(ofMillis(500)))
+                .aggregate(MockInitializer.STRING_INIT, sessionMerger, Materialized.with(Serdes.String(), Serdes.String()));
+        customers.toStream().to(OUTPUT);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+            final TestInputTopic<String, String> testInputTopic = driver.createInputTopic(
+                    TOPIC, new StringSerializer(), new StringSerializer());
+            final TestOutputTopic<Windowed<String>, String> testOutputTopic = driver.createOutputTopic(
+                    OUTPUT, new SessionWindowedDeserializer<>(new StringDeserializer()), new StringDeserializer());
+            testInputTopic.pipeInput("k1", "A", 0);
+            testInputTopic.pipeInput("k2", "A", 0);
+            testInputTopic.pipeInput("k1", "B", 599);
+            testInputTopic.pipeInput("k2", "B", 607);
+
+            assertOutputKeyValueTimestamp(testOutputTopic, "k1", "0+A", 0);
+            assertOutputKeyValueTimestamp(testOutputTopic, "k2", "0+A", 0);
+            assertOutputKeyValueTimestamp(testOutputTopic, "k1", "0+B", 599);
+            assertOutputKeyValueTimestamp(testOutputTopic, "k2", "0+B", 607);
+        }
+    }
+
+    @Test
+    public void sessionWindowAggregate2Test() {
+        final KTable<Windowed<String>, String> customers = groupedStream.cogroup(MockAggregator.TOSTRING_ADDER)
+                .windowedBy(SessionWindows.with(ofMillis(500)))
+                .aggregate(MockInitializer.STRING_INIT, sessionMerger, Materialized.with(Serdes.String(), Serdes.String()));
+        customers.toStream().to(OUTPUT);
```

Comment:
This seems to be the same program as in the test above -- if yes, I think we should merge both tests into one.